src/clj/puppetlabs/puppetserver/shell_utils.clj

```diff
@@ -0,0 +1,56 @@
+(ns puppetlabs.puppetserver.shell-utils
+  (:require [schema.core :as schema]
+            [clojure.java.io :as io])
+  (:import (com.puppetlabs.puppetserver ShellUtils)
+           (java.io IOException)))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Schemas
+
+(def ExecutionResult
+  "A map that contains the details of the result of executing a command."
+  {:exit-code schema/Int
+   :stderr schema/Str
+   :stdout schema/Str})
```

Comment:
At some point we will need a way to get `Stream` objects back out of stderr/stdout, instead of Strings, but that doesn't need to be in scope for this PR as long as you can see a path forward.  I can imagine possibly an alternate function that you'd call to initiate the execution, and a different schema that it returns.